### DISCUSSION TITLE
Sonoff mini switch1 rule

### DISCUSSION THREAD

**Discussion by polve93** [arendst/Tasmota#16948] [Q&A]
When i trigger a button (switch1), i want the load (power1) switch ON for 20 minutes and then switch OFF.

How do i should write the rule?
Thanks in advance

**Comment by barbudor** [comment]
[Pulsetime](https://tasmota.github.io/docs/Commands/#pulsetime) is simpler for that usage

**Reply by polve93** [reply]
> [Pulsetime](https://tasmota.github.io/docs/Commands/#pulsetime) is simpler for that usage

Pulsetime works well if there is a switch as input, in my case i have a button (triggers the input ON for a short amount of time, then goes OFF). Does Pulsetime work with buttons also?

**Reply by barbudor** [reply]
Pulsetime doesn't care if button or switch. It starts counting on the power ON event whereas it comes from button, switch, mqtt or web button.

Actually if you want the timer to be only from the button, then a rule would be the solution

**Reply by polve93** [reply]
> Pulsetime doesn't care if button or switch. 

See the [image](https://i.imgur.com/k1s1BWg.png) attached. Maybe i'm doing something wrong but Pulsetime is not working as i expect when a button trigger the input. 

> Actually if you want the timer to be only from the button, then a rule would be the solution

Yes, i need a rule, this is what i thought in the first message. This will be my first rule with tasmota, so any kind of help is appreciated.

**Reply by barbudor** [reply]
I see nothing wrong in your exemple
You are configuring pulsetime for 3 seconds and it power off after 3 seconds

For a rule that would trigger only on button and not on web button or mqtt:
```
Rule1 on button#state do backlog power on; RuleTimer1 1200 endon on rules#timer=1 do power off endoff
Rule1 1
```
